exposed-core/src/main/kotlin/org/jetbrains/exposed/dao/id/CompositeID.kt

```diff
@@ -0,0 +1,63 @@
+package org.jetbrains.exposed.dao.id
+
+import org.jetbrains.exposed.sql.Column
+
+/** Class representing a mapping of each composite primary key column to its stored identity value. */
+class CompositeID private constructor() : Comparable<CompositeID> {
+    internal val values: MutableMap<Column<*>, Comparable<*>?> = HashMap()
+
+    @Suppress("UNCHECKED_CAST")
+    @JvmName("setWithEntityIdValue")
+    operator fun <T : Comparable<T>, ID : EntityID<T>> set(column: Column<ID>, value: T) {
+        require(values.isEmpty() || values.keys.first().table == column.table) {
+            "CompositeID key columns must all come from the same IdTable"
+        }
+        values[column] = EntityID(value, column.table as IdTable<T>)
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    @JvmName("setWithNullableEntityIdValue")
+    operator fun <T : Comparable<T>, ID : EntityID<T>> set(column: Column<ID?>, value: T?) {
+        require(column.columnType.nullable || value != null) {
+            "Trying to set null to not nullable column $column"
+        }
+        values[column] = value?.let { EntityID(value, column.table as IdTable<T>) }
+    }
```

Comment:
It seems odd that a key column may be nullable, but this functionality exists for `EntityID` columns in other `IdTable`s, so I included it.